File: Omega/src/com/saggitt/omega/smartspace/SmartspacePreferencesShortcut.java
package com.saggitt.omega.smartspace;

import android.content.Intent;
import android.view.View;

import com.android.launcher3.Launcher;
import com.android.launcher3.R;
import com.android.launcher3.userevent.nano.LauncherLogProto;
import com.android.launcher3.views.OptionsPopupView;
import com.saggitt.omega.settings.SettingsActivity;

public class SmartspacePreferencesShortcut extends OptionsPopupView.OptionItem {

    public SmartspacePreferencesShortcut() {
        super(R.string.customize, R.drawable.ic_smartspace_preferences, LauncherLogProto.ControlType.SETTINGS_BUTTON,
                SmartspacePreferencesShortcut::startSmartspacePreferences);
    }

    private static boolean startSmartspacePreferences(View view) {
        Launcher launcher = Launcher.getLauncher(view.getContext());
        launcher.startActivitySafely(view, new Intent(launcher, SettingsActivity.class)
                .putExtra(SettingsActivity.SubSettingsFragment.TITLE, launcher.getString(R.string.home_widget))
                .putExtra(SettingsActivity.SubSettingsFragment.CONTENT_RES_ID, R.xml.omega_preferences_smartspace)
                .putExtra(SettingsActivity.SubSettingsFragment.HAS_PREVIEW, true), null, null);
        return true;
    }
}

File: Omega/src/com/saggitt/omega/smartspace/SmartSpacePreferencesShortcut.kt
package com.saggitt.omega.smartspace

import android.content.Context
import android.view.View
import android.view.View.OnLongClickListener
import com.android.launcher3.R
import com.android.launcher3.logging.StatsLogManager.EventEnum
import com.android.launcher3.views.OptionsPopupView.OptionItem
import com.saggitt.omega.compose.PrefsActivityX
import com.saggitt.omega.compose.navigation.Routes

class SmartSpacePreferencesShortcut(context: Context, eventId: EventEnum?) : OptionItem(
    context.getString(R.string.customize),
    context.getDrawable(R.drawable.ic_smartspace_preferences),
    eventId,
    OnLongClickListener { view: View -> startSmartspacePreferences(context) }) {

    companion object {
        private fun startSmartspacePreferences(context: Context): Boolean {
            context.startActivity(
                PrefsActivityX.createIntent(
                    context,
                    "${Routes.PREFS_WIDGETS}/"
                )
            )

            return true
        }
    }
}